### DISCUSSION TITLE
[@types/jest] - Mismatch between types in jest.expect Asymmetric Matcher

### DISCUSSION THREAD

**Discussion by aldirrix** [DefinitelyTyped/DefinitelyTyped#65862] [Issues with a @types package]
We use `@types/jest` to have the benefit of being able to acces the types for Jest globals without a need to import them but recently we wanted to use an [Asymmetric matcher](https://jestjs.io/docs/expect#asymmetric-matchers) and found that when we try the following:

```ts
expect({ foo: 'bar', decimalNumber: 0.01 }).toEqual({
  foo: 'bar', 
  decimalNumber: expect.closeTo(aDecimalNumber),
});
```

We get a type error `Unsafe assignment of an 'any' value` because the type definition states the following:`(method) jest.Expect.closeTo(num: number, numDigits?: number | undefined): any`

However, we can solve this issue if we add the import to the file with 
```ts
import { expect } from '@jest/globals';
```

This expect async matcher has a different signature: `(method) AsymmetricMatchers.closeTo(sample: number, precision?: number | undefined): AsymmetricMatcher_2`

This applies to all of the expect asym matchers and was wondering if this was by design

[a comment by an automated account is not shown]

**Comment by mrazauskas** [comment]
Thanks for reminder.

I can’t find it quickly, but there was a discussion about the return type of asymmetric matchers. This is easy to fix, but that is rather breaking for current users. So the fix has to wait till the next major.

**Reply by aldirrix** [reply]
Thanks for the response

I see, I also couldn't find any discussion related to this topic so it's good to know that this is already known and fix is easy and coming in the next major version, thanks for the input.
